Q:

Cannot launch nao_driver

[Go to Edit2 for the current error]
Hello, I am trying to launch nao_driver with the command:
NAO_IP=10.51.81.15 roslaunch nao_driver nao_driver.launch

or
NAO_IP=10.51.81.15 FORCE_PYTHON=true roslaunch nao_driver nao_driver.launch

but i get the following errors.
MyName@Wael:~$ NAO_IP=10.51.81.15 roslaunch nao_driver nao_driver.launch
... logging to /home/MyName/.ros/log/79d95dc2-8379-11e3-8cbe-90b11c6e6bd9/roslaunch-Wael-19571.log
Checking log directory for disk usage. This may take awhile.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server Wael:38416/

SUMMARY
========

PARAMETERS
 * /rosdistro
 * /rosversion

NODES
  /
    nao_behaviors (nao_driver/nao_behaviors.py)
    nao_controller (nao_driver/nao_controller.py)
    nao_leds (nao_driver/nao_leds.py)
    nao_sensors (nao_driver/nao_sensors_cpp)
    nao_speech (nao_driver/nao_speech.py)
    nao_tactile (nao_driver/nao_tactile.py)
    nao_walker (nao_driver/nao_walker.py)

auto-starting new master
process[master]: started with pid [19585]
ROS_MASTER_URI=localhost:11311

setting /run_id to 79d95dc2-8379-11e3-8cbe-90b11c6e6bd9
process[rosout-1]: started with pid [19598]
started core service [/rosout]
ERROR: cannot launch node of type [nao_driver/nao_sensors_cpp]: cant locate node [nao_sensors_cpp] in package [nao_driver]
process[nao_walker-3]: started with pid [19610]
process[nao_controller-4]: started with pid [19611]
process[nao_behaviors-5]: started with pid [19612]
process[nao_leds-6]: started with pid [19615]
process[nao_speech-7]: started with pid [19618]
process[nao_tactile-8]: started with pid [19623]
failed to load symbols, rosdep will not function properly
/opt/ros/hydro/lib/rosout/rosout: symbol lookup error: /opt/ros/hydro/lib/libroscpp.so: undefined symbol: _ZN3ros6HeaderC1Ev
[WARN] [WallTime: 1390404352.268687] Inbound TCP/IP connection failed: connection from sender terminated before handshake header received. 0 bytes were received. Please check sender for additional details.
[rosout-1] process has died [pid 19598, exit code 127, cmd /opt/ros/hydro/lib/rosout/rosout __name:=rosout __log:=/home/MyName/.ros/log/79d95dc2-8379-11e3-8cbe-90b11c6e6bd9/rosout-1.log].
log file: /home/MyName/.ros/log/79d95dc2-8379-11e3-8cbe-90b11c6e6bd9/rosout-1*.log
respawning...
[rosout-1] restarting process
process[rosout-1]: started with pid [19751]
/opt/ros/hydro/lib/rosout/rosout: symbol lookup error: /opt/ros/hydro/lib/libroscpp.so: undefined symbol: _ZN3ros6HeaderC1Ev
[WARN] [WallTime: 1390404352.577597] Inbound TCP/IP connection failed: connection from sender terminated before handshake header received. 0 bytes were received. Please check sender for additional details.
[rosout-1] process has died [pid 19751, exit code 127, cmd /opt/ros/hydro/lib/rosout/rosout __name:=rosout __log:=/home/MyName/.ros/log/79d95dc2-8379-11e3-8cbe-90b11c6e6bd9/rosout-1.log].
log file: /home/MyName/.ros/log/79d95dc2-8379-11e3-8cbe-90b11c6e6bd9/rosout-1*.log
respawning...
[rosout-1] restarting process

Any help would be greatly appreciated.
Edit:
Thank you AHornung, that seems to help a little. The errors has changed, but are still similar to what they were.
setting /run_id to f2362376-837f-11e3-b5cf-90b11c6e6bd9
process[rosout-1]: started with pid [20614]
started core service [/rosout]/
process[nao_sensors-2]: started with pid [20626]
process[nao_walker-3]: started with pid [20627]
process[nao_controller-4]: started with pid [20628]
process[nao_behaviors-5]: started with pid [20629]
process[nao_leds-6]: started with pid [20634]
Traceback (most recent call last):
  File "/opt/ros/hydro/lib/nao_driver/nao_sensors.py", line 44, in <module>
    from tf import transformations
  File "/opt/ros/hydro/lib/python2.7/dist-packages/tf/__init__.py", line 28, in <module>
    from _tf import *
ImportError: /opt/ros/hydro/lib/libroscpp.so: undefined symbol: _ZN3ros6HeaderD1Ev
process[nao_speech-7]: started with pid [20639]
process[nao_tactile-8]: started with pid [20646]
================================================================================REQUIRED process [nao_sensors-2] has died!
process has died [pid 20626, exit code 1, cmd /opt/ros/hydro/lib/nao_driver/nao_sensors.py --pip=10.51.81.15 --pport=9559 __name:=nao_sensors __log:=/home/antoinerioux/.ros/log/f2362376-837f-11e3-b5cf-90b11c6e6bd9/nao_sensors-2.log].
log file: /home/antoinerioux/.ros/log/f2362376-837f-11e3-b5cf-90b11c6e6bd9/nao_sensors-2*.log
Initiating shutdown!
================================================================================
[nao_tactile-8] killing on exit
[nao_speech-7] killing on exit
[nao_leds-6] killing on exit
[nao_behaviors-5] killing on exit
Traceback (most recent call last):
  File "/opt/ros/hydro/lib/nao_driver/nao_tactile.py", line 35, in <module>
    import rospy
  File "/opt/ros/hydro/lib/python2.7/dist-packages/rospy/__init__.py", line 49, in <module>
    from .client import spin, myargv, init_node, \
  File "/opt/ros/hydro/lib/python2.7/dist-packages/rospy/client.py", line 67, in <module>
    [nao_controller-4] killing on exit
from roscpp.srv import GetLoggers, GetLoggersResponse, SetLoggerLevel, SetLoggerLevelResponse
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roscpp/srv/__init__.py", line 2, in <module>
    from ._GetLoggers import *
[nao_walker-3] killing on exit
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roscpp/srv/_GetLoggers.py", line 92, in <module>
    import roscpp.msg
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roscpp/msg/__init__.py", line 1, in <module>
    from ._Logger import *
  File "/opt/ros/hydro/lib/python2.7/dist-packages/roscpp/msg/_Logger.py", line 1, in <module>
    """autogenerated by genpy from roscpp/Logger.msg. Do not edit."""
KeyboardInterrupt
[nao_sensors-2] killing on exit
failed to load symbols, rosdep will not function properly
Traceback (most recent call last):
  File "/opt/ros/hydro/lib/nao_driver/nao_leds.py", line 173, in <module>
    node = NaoLeds()
  File "/opt/ros/hydro/lib/nao_driver/nao_leds.py", line 58, in __init__
    rospy.init_node( self.NODE_NAME )
  File "/opt/ros/hydro/lib/python2.7/dist-packages/rospy/client.py", line 317, in init_node
    raise rospy.exceptions.ROSInitException("init_node interrupted before it could complete")
rospy.exceptions.ROSInitException: init_node interrupted before it could complete
Traceback (most recent call last):
  File "/opt/ros/hydro/lib/nao_driver/nao_behaviors.py", line 139, in <module>
    node = NaoBehaviors()
  File "/opt/ros/hydro/lib/nao_driver/nao_behaviors.py", line 59, in __init__
    rospy.init_node( self.NODE_NAME )
  File "/opt/ros/hydro/lib/python2.7/dist-packages/rospy/client.py", line 317, in init_node
    raise rospy.exceptions.ROSInitException("init_node interrupted before it could complete")
rospy.exceptions.ROSInitException: init_node interrupted before it could complete
/opt/ros/hydro/lib/rosout/rosout: symbol lookup error: /opt/ros/hydro/lib/libroscpp.so: undefined symbol: _ZN3ros6HeaderC1Ev
[WARN] [WallTime: 1390407131.509740] Inbound TCP/IP connection failed: connection from sender terminated before handshake header received. 0 bytes were received. Please check sender for additional details.
[nao_speech-7] escalating to SIGTERM
[rosout-1] killing on exit
[master] killing on exit
shutting down processing monitor...
... shutting down processing monitor complete
done

Edit2:
I resolved the problem above with the suggestion on another question at:
answers.ros.org/question/120541/problem-of-running-rosrun-turtlesim-turtlesim_node/
But now I have yet another error!
    antoinerioux@Wael:~$ roslaunch nao_driver nao_driver.launch force_python:=true nao_ip:=10.51.81.15
... logging to /home/antoinerioux/.ros/log/e05386e2-8449-11e3-8e29-90b11c6e6bd9/roslaunch-Wael-16790.log
Checking log directory for disk usage. This may take awhile.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server Wael:59162/

SUMMARY
========

PARAMETERS
 * /rosdistro
 * /rosversion

NODES
  /
    nao_behaviors (nao_driver/nao_behaviors.py)
    nao_controller (nao_driver/nao_controller.py)
    nao_leds (nao_driver/nao_leds.py)
    nao_sensors (nao_driver/nao_sensors.py)
    nao_speech (nao_driver/nao_speech.py)
    nao_tactile (nao_driver/nao_tactile.py)
    nao_walker (nao_driver/nao_walker.py)

auto-starting new master
process[master]: started with pid [16804]
ROS_MASTER_URI=//localhost:11311

setting /run_id to e05386e2-8449-11e3-8e29-90b11c6e6bd9
process[rosout-1]: started with pid [16817]
started core service [/rosout]/
process[nao_sensors-2]: started with pid [16829]
process[nao_walker-3]: started with pid [16830]
process[nao_controller-4]: started with pid [16831]
process[nao_behaviors-5]: started with pid [16832]
process[nao_leds-6]: started with pid [16837]
process[nao_speech-7]: started with pid [16840]
process[nao_tactile-8]: started with pid [16841]
failed to load symbols, rosdep will not function properly
Traceback (most recent call last):
  File "/opt/ros/hydro/lib/nao_driver/nao_sensors.py", line 44, in <module>
    from tf import transformations
  File "/opt/ros/hydro/lib/python2.7/dist-packages/tf/__init__.py", line 29, in <module>
    from listener import TransformListener, TransformerROS
  File "/opt/ros/hydro/lib/python2.7/dist-packages/tf/listener.py", line 34, in <module>
    from tf import transformations
  File "/opt/ros/hydro/lib/python2.7/dist-packages/tf/transformations.py", line 174, in <module>
    import numpy
  File "/usr/share/pyshared/numpy/__init__.py", line 137, in <module>
    import add_newdocs
  File "/usr/share/pyshared/numpy/add_newdocs.py", line 9, in <module>
    from numpy.lib import add_newdoc
  File "/usr/share/pyshared/numpy/lib/__init__.py", line 4, in <module>
    from type_check import *
  File "/usr/share/pyshared/numpy/lib/type_check.py", line 8, in <module>
    import numpy.core.numeric as _nx
  File "/usr/share/pyshared/numpy/core/__init__.py", line 5, in <module>
    import multiarray
ImportError: No module named multiarray
failed to load symbols, rosdep will not function properly
================================================================================REQUIRED process [nao_sensors-2] has died!
process has died [pid 16829, exit code 1, cmd /opt/ros/hydro/lib/nao_driver/nao_sensors.py --pip=10.51.81.15 --pport=9559 __name:=nao_sensors __log:=/home/antoinerioux/.ros/log/e05386e2-8449-11e3-8e29-90b11c6e6bd9/nao_sensors-2.log].
log file: /home/antoinerioux/.ros/log/e05386e2-8449-11e3-8e29-90b11c6e6bd9/nao_sensors-2*.log
Initiating shutdown!
================================================================================
[nao_tactile-8] killing on exit
[nao_speech-7] killing on exit
[nao_leds-6] killing on exit
[nao_behaviors-5] killing on exit
[nao_controller-4] killing on exit
[nao_walker-3] killing on exit
[nao_sensors-2] killing on exit
Traceback (most recent call last):
  File "/opt/ros/hydro/lib/nao_driver/nao_tactile.py", line 192, in <module>
    ROSNaoTactileModule = NaoTactile("ROSNaoTactileModule")
  File "/opt/ros/hydro/lib/nao_driver/nao_tactile.py", line 80, in __init__
    self.init_almodule()
  File "/opt/ros/hydro/lib/nao_driver/nao_tactile.py", line 121, in init_almodule
    self.broker = ALBroker("%sBroker" % self.moduleName, self.ip, self.port, self.pip, self.pport)
  File "/home/antoinerioux/pynaoqi-python-2.7-naoqi-1.14-linux64/inaoqi.py", line 164, in __init__
    this = _inaoqi.new_broker(*args)
KeyboardInterrupt
Exception TypeError: "in method 'module_getName', argument 1 of type 'AL::module *'" in <bound method NaoTactile.__del__ of <__main__.NaoTactile;  >> ignored
[rosout-1] killing on exit
[master] killing on exit
shutting down processing monitor...
... shutting down processing monitor complete
done

Reinstalled Ubuntu and ROS. Everything worked on the first try after that when using the syntax proposed in the answer.

Originally posted by TopSecret on ROS Answers with karma: 37 on 2014-01-22
Post score: 0

A:

You need to run roslaunch nao_driver.launch force_python:=true
(with the optional NAO_IP in front). NAO_IP is an environment variable that you can globally set or directly set before launching, whereas force_python is a launch file argument (you can also set nao_ip=... as a launch file argument as alternative but vice-versa there is no environment variable called FORCE_PYTHON used).
That import error looks like a general problem with ROS:
ImportError: /opt/ros/hydro/lib/libroscpp.so: undefined symbol: _ZN3ros6HeaderD1Ev

Are other nodes in ROS (specifically Python nodes) working fine, did you manually change things in /opt/ros/hydro/...?

Originally posted by AHornung with karma: 5904 on 2014-01-22
This answer was ACCEPTED on the original site
Post score: 1

Original comments
Comment by TopSecret on 2014-01-22:
Since I just started with ROS + Nao + Ubuntu, I wouldn't really know what other nodes to try for verification. I am mostly following tutorials at the moment. I don't remember changing anything in this directory.
Comment by AHornung on 2014-01-22:
On what operating system is that happening? Did you freshly install ROS from the repositories?
Comment by AHornung on 2014-01-22:
Also, did you source /opt/ros/hydro/setup.bash
Comment by TopSecret on 2014-01-22:
I solved the symbol problem following this answer: http://answers.ros.org/question/120541/problem-of-running-rosrun-turtlesim-turtlesim_node/.

